tests/catalog/test_db.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+from sqlalchemy import create_engine
+from sqlalchemy.orm import Session
+from datetime import datetime
+from recap.catalogs.db import DatabaseCatalog, CatalogEntry
+
+class TestCatalogEntry:
+    def test_is_deleted(self):
+        entry = CatalogEntry(deleted_at=datetime.now())
+        assert entry.is_deleted() == True
+
+        entry = CatalogEntry(deleted_at=None)
+        assert entry.is_deleted() == False
+
+class TestDatabaseCatalog:
+    @pytest.fixture
+    def engine(self):
+        return create_engine('sqlite:///:memory:')
+
+    @pytest.fixture
+    def catalog(self, engine):
+        return DatabaseCatalog(engine)
+
+    def test_catalog_init(self, catalog):
+        assert isinstance(catalog.Session(), Session)
+
+    def test_catalog_touch(self, catalog):
+        # Test touching a path that doesn't exist
+        catalog.touch("/databases/table")
+        with catalog.Session() as session:
+            result = session.query(CatalogEntry).filter_by(parent="/databases", name="table").first()
+            assert result.metadata_ == {}
+
+        # Test touching a path that does exist
+        catalog.touch("/databases/table")
+        with catalog.Session() as session:
+            result = session.query(CatalogEntry).filter_by(parent="/databases", name="table").first()
+            assert result.metadata_ == {}
+
+        # Test touching a path that has been deleted
+        with catalog.Session() as session:
+            entry = session.query(CatalogEntry).filter_by(parent="/databases", name="table").first()
+            entry.deleted_at = datetime.now()
+
+        catalog.touch("/databases/table")
+        with catalog.Session() as session:
+            result = session.query(CatalogEntry).filter_by(parent="/databases", name="table").first()
+            assert result.metadata_ == {}
+
+    def test_write(self, catalog):
+        metadata = {
+            "db.location": {
```

Comment:
Can you use the actual Pydantic `Location` model here?

Reply:
Done